### PR TITLE
Migrate to Github Actions

### PR DESCRIPTION
**Pull request by Mshahidtaj**
#### Summary
Migrate to Github Actions

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-5087

### CODE REVIEW

File: .github/workflows/.test.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+jobs:
+  check-style:
+    runs-on: ubuntu-latest
+    container:
+      image: golang:1.18.0-bullseye
+    steps:
+    - name: ci/checkout-repo
+      uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+    - name: ci/cache-gopath
```
Comment (stafot) [inline .github/workflows/.test.yml:13]: NIce one!

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          fetch-depth: 0
+      - name: ci/publish-release-notes
+        run: |
+          make deps
+          make release
+
```
Comment (stafot) [inline .github/workflows/cd.yml]: @Mshahidtaj Please lint these to have newline in the end

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+jobs:
+  test:
+    uses: ./.github/workflows/.test.yml
+  push-docker:
+    uses: ./.github/workflows/.docker-push.yml
+    with:
+      is_pr: "${{ github.ref != 'refs/heads/main' }}"
+    secrets: inherit
+    needs: [test]
```
Comment (stafot) [inline .github/workflows/ci.yml]: As above

**Review by stafot — APPROVED** [review]
Check my comments for the minor changes that are needed. Great work overall

File: .github/actions/docker-prepare/action.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+name: 'docker-prepare'
+description: 'Install docker requirements'
+
+runs:
+  using: 'composite'
+  steps:
+  - name: ci/setup-buildx
+    uses: docker/setup-buildx-action@15c905b16b06416d2086efa066dd8e3a35cc7f98 # v2.4.0
+    with:
+      version: v0.7.1
```
Comment (stafot) [inline .github/actions/docker-prepare/action.yml]: @Mshahidtaj Please lint these to have newline in the end

File: .github/workflows/.docker-push.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    - name: ci/checkout-repo
+      uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+    - name: ci/prepare-docker-environment
+      uses: ./.github/actions/docker-prepare
+    - name: ci/push-image-pr
+      run: "make push-image-pr"
+      if: "${{ inputs.is_pr }}"
+    - name: ci/push-image
+      run: "make push-image"
+      if: "${{ !inputs.is_pr }}"
```
Comment (stafot) [inline .github/workflows/.docker-push.yml]: @Mshahidtaj Please lint these to have newline in the end

File: Makefile [hunk 6]
```diff
@@ -62,16 +63,57 @@ build: ## Build the mattermost-apps-cloud-deployer
...
+	--platform linux/arm64,linux/amd64 \
 	--build-arg DOCKER_BUILD_IMAGE=$(DOCKER_BUILD_IMAGE) \
 	--build-arg DOCKER_BASE_IMAGE=$(DOCKER_BASE_IMAGE) \
 	. -f build/Dockerfile -t $(MATTERMOST_APPS_CLOUD_DEPLOYER_IMAGE) \
-	--no-cache
+	--no-cache \
+	--push
+
+.PHONY: build-image-with-tag
+build-image-with-tag:  ## Build the docker image for mattermost-cloud-database-factory
```
Comment (stafot) [inline Makefile]: Minor wording

**Review by angeloskyratzakos — APPROVED** [review]

**Pull request merged**